internal/builders/docker/commands.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 SLSA Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+// This file contains definitions of the subcommands of the
+// `slsa-docker-based-generator` command.
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/slsa-framework/slsa-github-generator/internal/builders/docker/pkg"
+	"github.com/spf13/cobra"
+)
+
+// DryRunCmd validates the input flags, generates a BuildDefinition from them.
+func DryRunCmd(check func(error)) *cobra.Command {
+	var buildConfigPath string
+	var sourceRepo string
+	var gitCommitHash string
+	var builderImage string
+	var buildDefinitionPath string
+
+	var cmd = &cobra.Command{
+		Use:   "dry-run [FLAGS]",
+		Short: "Generates and stores a JSON-formatted BuildDefinition based on the input arguments.",
+		Run: func(cmd *cobra.Command, args []string) {
+			// TODO(#1191): Parse the input arguments into an instance of BuildDefinition.
+			bd := &pkg.BuildDefinition{}
+			check(writeBuildDefinitionToFile(*bd, buildDefinitionPath))
+		},
+	}
+
+	cmd.Flags().StringVarP(&buildConfigPath, "build_config_path", "c", "", "Required - Path to a toml file containing the build configs.")
```

Comment:
since these are in common with the build command, you could group these as input options. lemme send a quick patch

Reply:
https://github.com/slsa-framework/slsa-github-generator/commit/9c7bf30bd6e5544455f2df4bf68e42d52a33750c

check this out!

Reply:
Nice. Thanks. I'll apply thin on my fork.

Reply:
Done.